draft-thomson-httpbis-h2-stream-limits.md

```diff
@@ -53,18 +57,23 @@ of abuse.
 
 # Introduction
 
-HTTP/2 {{RFC9113}} allows endpoints to describe a concurrency limit for streams
-(`SETTINGS_MAX_CONCURRENT_STREAMS`).  Initially, the stream concurrency limit is
-only bounded by the maximum number of streams that can be created, which is
-2<sup>30</sup> for each endpoint, but the value can be changed at any time.
-Most endpoints set a smaller value in an attempt to protect the resources that
-are committed when processing a stream.
-
-This limit is not effective in the case that streams are quickly cancelled.
-Stream cancellation with the `RST_STREAM` frame has immediate effect, which
-means that the stream no longer counts against the concurrent stream limit.
-This means that a malicious endpoint can create and cancel an unbounded number
-of streams as long as its peer does not set a limit of zero.
+HTTP/2 allows endpoints to declare the concurrency limit for streams created by
+their peer (`SETTINGS_MAX_CONCURRENT_STREAMS`; see {{Section 5.1.2 of
+RFC9113}}). Initially, the stream concurrency limit is only bounded by the
+maximum number of streams that can be created, which is 2<sup>30</sup> for each
+endpoint, but the value can be changed at any time. Most endpoints set a smaller
+value in an attempt to protect the resources that are committed when processing
+a stream.
+
+This limit is not effective in the case that streams are quickly cancelled or
+closed. The creator of a stream can cancel it using the `RST_STREAM` frame.  The
+creator of a stream can also purposefully send frames that violate HTTP/2 rules,
+forcing a compliant receiver to close the stream. Either of these methods has
```

Comment:
Do you think that it is worth noting that a stream error like this could result in a connection error, which would make the error -> close/reset path unavailable to an attacker.  Obviously, an attacker can simply check what the victim would do before trying that approach.

Reply:
That crossed my mind yes. Even if a client didn't check, there's not much punishment for just giving it a go - it either works or it doesn't and the resources (e.g. bandwidth) required to try it out are not enormous.

An endpoint might even close _some_ streams until a threshold is reached - and then close the connection. So trying to explain that felt like it could be a distraction.

I can add a sentence here if you like, or punt it to the security considerations, or do nothing.